tests/bins_test.py

```diff
@@ -212,6 +212,13 @@ def test_bins_view_coords_drop():
     assert set(var.bins.coords) == {'time'}
 
 
+def test_bins_view_coords_assign():
+    var = make_binned()
+    assert set(var.bins.coords) == {'time'}
+    new = var.bins.assign_coords(('time',))
```

Comment:
That is not how `assign_coords` works, you need to pass a variable. Please have a look at the existing implementations or tests for examples.